### PR TITLE
2023 refactor for the large Testimonial

### PR DESCRIPTION
**Pull request by mperrotti**
## Summary

Updates the design of the Testimonial component to match the 2023 refactor specs.

## List of notable changes:

- makes the quote mark proportional to the quote text
- adds the ability to highlight a portion of the quote text
- makes the quote mark color customizable

## What should reviewers focus on?

Does it look right?

## Steps to test:

1. Go to the Storybook
2. Go to the "Highlighted Portion" story under "Testimonial"

## Supporting resources (related issues, external links, etc):

- [Figma](https://www.figma.com/file/wuhr1sqnv2YEaT4F6dUBbl/2023-Refactor?type=design&node-id=2589-197735&mode=design&t=s2GBuKqMluCLpxFY-4)
- [Primer Web branch](https://www.figma.com/file/BJ95AjraesmRCWsKA013GS/branch/DasZxE7UgaNFcJO2SsTDQA/Primer-Brand?type=design&node-id=526%3A8841&mode=design&t=wHVGsZv8wqOfqlyg-1) (WIP)
- https://github.com/github/primer/issues/2506

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

> Please try to provide before and after screenshots or videos

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

Screenshot 2023-08-03 at 2 53 26 PM

 </td>
<td valign="top">

Screenshot 2023-08-03 at 2 55 08 PM

</td>
</tr>
</table>

### CODE REVIEW

**Comment by mperrotti** [comment]
Will be converted to "ready for review" once I get some questions answered: https://github.com/github/primer/issues/2506#issuecomment-1664443660

File: packages/react/src/Testimonial/Testimonial.tsx
Comment (mperrotti) [inline packages/react/src/Testimonial/Testimonial.tsx:31]: I used the word "Fill" incase we decide to allow gradient or image fills in the future.

File: packages/react/src/Testimonial/Testimonial.module.css
Comment (rezrah) [inline packages/react/src/Testimonial/Testimonial.module.css:44]: `has` doesn't have comprehensive browser support yet, so could you handle this inside the component for the time being? [Example of `em` support added to section intro using `child.type`](https://github.com/primer/brand/pull/376/files#diff-dcc5852ace9baf504d6890f39cf2a451591d81598cbdcc9bcb4f716c8a74f69fR51-R60).

File: packages/react/src/Testimonial/Testimonial.module.css
Comment (rezrah) [inline packages/react/src/Testimonial/Testimonial.module.css:2]: Does this need to be a token? If so, could we prefix these tokens with `--brand` please for consistency with our other tokens. Would be great to also hoist it into our dedicated design tokens package please. Here's where they live: https://github.com/primer/brand/tree/main/packages/design-tokens/src/tokens/functional/components. You will need to create a new json source file there.

File: packages/react/src/Testimonial/Testimonial.module.css
Comment (mperrotti) [reply packages/react/src/Testimonial/Testimonial.module.css:44]: Falling back to a single color seems reasonable for this. Curious what other folks think.

File: packages/design-tokens/src/tokens/functional/components/testimonial/base.json
Comment (rezrah) [inline packages/design-tokens/src/tokens/functional/components/testimonial/base.json:9]: 👋 @mperrotti this is currently setting at 48px, but it should be the new 40px we've introduced in global typography refactoring recently. Could we update to use those styles? There are other associated styles we would also want to set here, like line-height. For reference 👇 

```
.Heading--4 {
  font-weight: var(--brand-heading-weight-600);
  font-size: var(--brand-text-size-600);
  line-height: var(--brand-text-lineHeight-600);
  font-family: var(--brand-heading-fontFamily);
  letter-spacing: var(--brand-heading-letterSpacing-600);
}
```

File: packages/design-tokens/src/tokens/functional/components/testimonial/base.json
Comment (rezrah) [inline packages/design-tokens/src/tokens/functional/components/testimonial/base.json:14]: The formatting seems off in this file, could you run a prettier fix on it please? Possible why the CI checks are failing too.

File: packages/react/src/Testimonial/Testimonial.module.css
Comment (rezrah) [inline packages/react/src/Testimonial/Testimonial.module.css:11]: This doesn't appear to be assigned to a value, but should be customisable 👍 

Could you hoist this into a colors.json design token source file please?

File: packages/react/src/Testimonial/Testimonial.stories.tsx
Comment (rezrah) [inline packages/react/src/Testimonial/Testimonial.stories.tsx:12]: FYI 👇 

Screenshot 2023-08-15 at 10 44 47

Could we set the default value in the design token layer also?

```suggestion
    quoteMarkFill: 'var(--brand-color-text-default)',
```

File: packages/react/src/Testimonial/Testimonial.module.css
Comment (mperrotti) [reply packages/react/src/Testimonial/Testimonial.module.css:11]: I'm wondering if it makes more sense to assign a default by providing a default value to the `quoteMarkFill` prop. Let me know which you prefer.

File: .changeset/modern-kings-live.md
Comment (rezrah) [inline .changeset/modern-kings-live.md:5]: Just because that's an internal codename...

```suggestion
Updated Testimonial component visuals, and enables customization of the quote color.
```


Might also be nice to show an example of the quote fill prop and the `<em>` thing.

File: packages/react/src/Testimonial/Testimonial.module.css
Comment (rezrah) [reply packages/react/src/Testimonial/Testimonial.module.css:14]: > I'm wondering if it makes more sense to assign a default by providing a default value to the quoteMarkFill prop

We usually set all default values and design logic variables in the primitive layer first.

Also fine if you want to set it in the React layer for now, as it can always be changed later 👍

File: packages/react/src/Testimonial/Testimonial.tsx
Comment (rezrah) [inline packages/react/src/Testimonial/Testimonial.tsx:124]: Assuming this merges after the river breakout, would be great to replace this with the helper component you introduced over there.

**Review by rezrah — APPROVED** [review]
Looks great 🙌 

Two things to check before we merge:

- Could you add the `quoteMarkFill` prop to the docs prop table please?
Screenshot 2023-08-18 at 10 05 11

- could you regen the snapshots too by adding the label again? you need to remove the label first, then re-add. It seems like your visual diffs are off.

**Review by josepmartins — APPROVED** [review]
Looks good overall! I just left a comment on the `quoteMarkFill` prop, but approving this to not block the PR. 

It would be nice to have an example using the <em> on large testimonials quotes in the React docs too 👍 .

File: packages/react/src/Testimonial/Testimonial.stories.tsx
Comment (mperrotti) [reply packages/react/src/Testimonial/Testimonial.stories.tsx:26]: I'm using `'text'` to support CSS variables. Happy to change to color though if you don't think it's worth it.

File: packages/react/src/Testimonial/Testimonial.tsx
Comment (mperrotti) [reply packages/react/src/Testimonial/Testimonial.tsx:42]: Yea, I think this is probably the right way to go. I still prefer `quoteMarkFill` over `quoteMarkColor`, but it's better to be consistent with the naming convention established with Label.

File: packages/react/src/Testimonial/Testimonial.stories.tsx
Comment (josepmartins) [reply packages/react/src/Testimonial/Testimonial.stories.tsx:26]: ah that makes sense. Another option if we choose to provide restricted colors is a list of inputs, like we do on Label playground

https://github.com/primer/brand/blob/main/packages/react/src/Label/Label.stories.tsx#L18

Comment (mperrotti) [reply packages/react/src/Testimonial/Testimonial.stories.tsx:26]: If we're only going to allow a limited set, I'd actually change this to a select dropdown or radio buttons.

File: packages/react/src/Testimonial/Testimonial.tsx
Comment (rezrah) [reply packages/react/src/Testimonial/Testimonial.tsx:42]: > Shouldn't we provide a restricted list of color/gradients options as well as we do in the Label component?

I think some presets would be useful, agree. We also need to handle use-cases where the value can be overridden from a CMS to be any arbitrary value. Example being the upcoming Universe pages, which feature unique accents. If we can make it so one of the supported colors is an `accent` type, then it makes it easier to override it.

File: packages/react/src/Testimonial/Testimonial.tsx
Comment (josepmartins) [inline packages/react/src/Testimonial/Testimonial.tsx:139]: we could just change the variant of the Text component, like we do in `Timeline`

https://github.com/primer/brand/blob/main/packages/react/src/Timeline/Timeline.tsx#L45

File: packages/react/src/Testimonial/Testimonial.module.css
Comment (josepmartins) [inline packages/react/src/Testimonial/Testimonial.module.css:51]: referring to this [comment](https://github.com/primer/brand/pull/372/files#r1298600302) we might not need this styles. 

Anyways the color should be `muted`, as well as the SectionIntro or the Timeline `em` color implementations

```suggestion
.Testimonial-quote--muted {
  color: var(--brand-color-text-muted);
}
```

File: packages/react/src/Testimonial/Testimonial.tsx
Comment (mperrotti) [reply packages/react/src/Testimonial/Testimonial.tsx]: I just saw this and didn't add an `accent` type. What would the value of `accent` be?

**Comment by jesussandreas** [comment]
LGTM!

have we consider updating the avatar section on these?

**Pull request merged**
